winterfell/webapp/src/js/profileService.js

```diff
@@ -1,12 +1,47 @@
 'use strict';
 var app = angular.module('vetafiApp');
-app.factory('profileService', ['claimService', function(claimService) {
-  return {
-    userInfo: {},
-    clearUserInfo: function() {
-      sessionStorageHelper.removePair(vfiConstants.keyUserId);
-      this.userInfo = {};
+app.service('Profile', ['claimService', '$q', 'net', function(claimService, $q, net) {
+    this.user = null;
+
+    this.logout = function() {
       claimService.clearClaims();
-    }
-  };
+      this.unsetUser();
```

Comment:
Redirecting to the logout page is what actually logs hte user our, because that is when the server tells the browser to unset the cookie. Because the express-session cookie is http only, we cant touch it from javascript, even to test if it exists. This just clears the service state for good measure